### PR TITLE
Remove keystroke and stopTyping extensions

### LINKED ISSUE #3146: Move `keystroke` and `stopTyping` extension to LLC
### Context & proposal

To remove `ChatDomain` completely and improve the public API of `ChatClient` we need to move these extension functions into LLC itself.

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] There should not be any extension function `ChatClient::keystroke` and `ChatClient::stopTyping`.
- [ ] The `keystroke` and `stopTyping` function should be inside low level `ChatClient`.
- [ ] `ChannelClient` must use this new function.

### PR DESCRIPTION
**Pull request by amitkma**
Closes #3146 
### 🎯 Goal

To remove ChatDomain completely and improve the public API of ChatClient we need to move these extension functions into LLC itself.

### 🛠 Implementation details

- `ChatClient::keystroke` and `ChatClient::stopTyping` extensions are removed.
- Those methods are now in the main `ChatClient` class.
- `TypingEventListener` is added to run the side-effects for typing events.
- ChannelController tests are migrated or removed. 

### 🧪 Testing
- Open sample app in two devices.
- Start typing in one device.
- You should be able to see `typing.start` or `typing.stop` events in logcat. (There is some issue with UI not being updated but that is out of the scope of this PR as this issue is already present in `develop`)

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt [hunk 1]
```diff
@@ -654,32 +653,22 @@ public class ChannelClient internal constructor(
 
     @CheckResult
     public fun keystroke(): Call<ChatEvent> {
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt:655]: Can we remove this method? We have the same one below 🤔

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt [hunk 2]
```diff
@@ -654,32 +653,22 @@ public class ChannelClient internal constructor(
...
 
     @CheckResult
-    public fun keystroke(parentId: String): Call<ChatEvent> {
-        return client.sendEvent(
-            eventType = EventType.TYPING_START,
-            channelType = channelType,
-            channelId = channelId,
-            extraData = mapOf(ARG_TYPING_PARENT_ID to parentId),
-        )
+    public fun keystroke(parentId: String? = null): Call<ChatEvent> {
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt:660]: Can we add KDoc here?

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt [hunk 3]
```diff
@@ -654,32 +653,22 @@ public class ChannelClient internal constructor(
...
+        return client.keystroke(channelType, channelId, parentId)
     }
 
     @CheckResult
     public fun stopTyping(): Call<ChatEvent> {
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt:665]: Same here - let's remove this one and make `parentId: String? = null`

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt [hunk 4]
```diff
@@ -654,32 +653,22 @@ public class ChannelClient internal constructor(
...
-        return client.sendEvent(EventType.TYPING_STOP, channelType, channelId)
+        return client.stopTyping(channelType, channelId)
     }
 
     @CheckResult
     public fun stopTyping(parentId: String): Call<ChatEvent> {
```
Comment (adasiewiczr) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt:670]: Can we add KDoc here?

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/listener/TypingEventListenerImpl.kt [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+import io.getstream.chat.android.client.experimental.plugin.listeners.TypingEventListener
+import io.getstream.chat.android.client.models.EventType
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.offline.experimental.channel.state.ChannelMutableState
+import io.getstream.chat.android.offline.experimental.channel.state.toMutableState
+import io.getstream.chat.android.offline.experimental.plugin.state.StateRegistry
+import java.util.Date
+
+@ExperimentalStreamChatApi
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/listener/TypingEventListenerImpl.kt:14]: Let's remove this annotation from every class in this PR

Comment (adasiewiczr) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/plugin/listener/TypingEventListenerImpl.kt:14]: Can we also add KDocs to this class?

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt (hunk 1 above)
Comment (amitkma) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt:655]: The other one takes `parentId`. Then we will have to add `@JvmOverloads`.

Comment (adasiewiczr) [reply stream-chat-android-client/src/main/java/io/getstream/chat/android/client/channel/ChannelClient.kt:655]: Let's keep only one and add `@JvmOverloads`

**Review by adasiewiczr — APPROVED** [review]
As discussed on Slack - typings are updated properly but the indicator is not being shown. The bug will be fixed in a separate PR.

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] All code we touched has new or updated KDocs

**Pull request merged**